airbyte-server/src/main/java/io/airbyte/server/handlers/BackwardCompatibilityHandler.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.server.handlers;
+
+import static io.airbyte.server.constants.DockerImageName.*;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.SourceConnection;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.util.Set;
+
+public class BackwardCompatibilityHandler {
```

Comment:
it would be great to cover this class with tests

Reply:
> it would be great to cover this class with tests

@alexandr-shegeda added unit tests